runtime/services/lightd/effects.js

```diff
@@ -69,6 +70,65 @@ function applyAlphaFactor (alpha) {
   return alpha
 }
 
+/**
+ * Color class for light
+ */
+class Color {
+  constructor (r, g, b, a) {
+    this.setRGBA(r, g, b, a)
+  }
+
+  /**
+   * set rgba
+   * @param {number} r
+   * @param {number} g
+   * @param {number} b
+   * @param {number} a
+   */
+  setRGBA (r, g, b, a) {
+    this._r = r
+    this._g = g
+    this._b = b
+    this._a = a
+  }
+  get r () {
+    return this._r
+  }
+  set r (val) {
+    this._r = val
+  }
+  get g () {
+    return this._g
+  }
+  set g (val) {
+    this._g = val
+  }
+  get b () {
+    return this._b
+  }
+  set b (val) {
+    this._b = val
+  }
+  get a () {
+    return this._a
+  }
+  set a (val) {
+    this._a = val
+  }
+
+  /**
+   * get alpha with factor
+   * @returns {number} alpha for render
+   */
+  get alphaWithFactor () {
+    if (this._a !== undefined && typeof this._a === 'number' && this._a >= 0 && this._a <= 1) {
+      return this._a * globalAlphaFactor
```

Comment:
It's better to maintain a copy of this value inside the class `Color`:

```js
color.setAlphaFactor(0.6)
color.toRGBA()
```
By the way, the color has multiple formats like RGB, RGBA, HSL and HSLA, thus I think the API should be:

```js
var color = new Color({ r: 255, g: 255, b: 255, a: 100 })
console.log(color.format) // rgba
console.log(color.toRGBA()) // [ 255, 255, 255, 100 ]

color.setAlphaFactor(0.7)
console.log(color.toRGBA()) // [ 255, 255, 255, 70 ]
```